sips/sip-tbd/sip-TBD-token-metadata-update-notifications.md

```diff
@@ -0,0 +1,255 @@
+```
+
+| Key                   | Value                                                                  |
+|-----------------------|------------------------------------------------------------------------|
+| `notification`        | The string `"token-metadata-update"`                                   |
+| `payload.token-class` | The string `"ft"`                                                      |
+| `payload.contract-id` | The contract id (principal) of the contract that declared the token    |
+
+## Non-Fungible Tokens
+
+When a contract needs to notify the network that metadata has changed for a **Non-Fungible Token**,
+it shall call `print` with a tuple with the following structure:
+
+```clarity
+{ notification: "token-metadata-update", payload: { token-class: "nft", token-ids: (list u100, u101), contract-id: <token contract id> }}
+```
+
+| Key                   | Value                                                                  |
+|-----------------------|------------------------------------------------------------------------|
+| `notification`        | The string `"token-metadata-update"`                                   |
+| `payload.token-class` | The string `"nft"`                                                     |
+| `payload.token-ids`   | A list with the uint token ids that need to be refreshed               |
+| `payload.contract-id` | The contract id (principal) of the contract that declared the tokens   |
+
+## Considerations for metadata indexers
+
+Metadata indexers should review the following implications when reacting to token metadata update
+notifications:
+
+* Notifications can come at any point in time and are persistent in the Stacks blockchain.
+  * When performing a local sync to the chain tip, old notifications for old metadata updates could
+    not necessarily have a distinct effect in metadata responses when processed in the present.
+* Multiple notifications for the same tokens will not necessarily correspond to multiple metadata
+  updates.
+  * Refreshing a token's metadata should be an idempotent operation. Repeated refreshes should not
+    create distinct records in the internal metadata database.
+* Notifications could be gratuitous.
+  * To prevent slow performance and guard against any Denial of Service attack attempts, contract
+    call rate limiting should be implemented locally.
+* Notifications can be delayed and out of order.
+  * A notification transaction's timestamp should not be considered to be the time when the token
+    metadata was actually updated.
+
+# Backwards compatibility
+
+Developers who need to emit metadata update notifications for tokens declared in older contracts
+(that were deployed before this notification standard was established) could do so by either calling
+the contract described in [Reference Implementations](#reference-implementations) or by deploying a
+new separate contract containing a public function that emits this notification.
```

Comment:
If the developer takes the latter option, is there a way for the indexer to discover this new contract, and pair it with the legacy contract?

Reply:
Good point. Just added some clarification to say they must first deploy and then call the new notifier contract so indexers can catch the updates.